MvvmCross/Platforms/Uap/Presenters/MvxWindowsViewPresenter.cs

```diff
@@ -240,5 +250,39 @@ protected virtual Task<bool> ShowPage(Type viewType, MvxBasePresentationAttribut
                 return Task.FromResult(false);
             }
         }
+
+        protected virtual async Task<bool> ShowModal(Type viewType, MvxBasePresentationAttribute attribute, MvxViewModelRequest request)
+        {
+            try
+            {
+                if (Activator.CreateInstance(viewType) is MvxWindowsContentDialog modalView)
+                {
+                    var instanceReques = (MvxViewModelInstanceRequest)request;
+                    modalView.ViewModel = instanceReques.ViewModelInstance;
+
+                    _windowsContentDialog = modalView;
+
+                    var modalAttribute = (MvxModalViewPresentationAttribute)attribute;
+                    await modalView.ShowAsync(modalAttribute.Placement);
+
+                    return true;
+                }
+
+                return false;
+            }
+            catch (Exception exception)
+            {
+                MvxLog.Instance.Trace("Error seen during navigation request to {0} - error {1}", request.ViewModelType.Name,
+                    exception.ToLongString());
+                return false;
+            }
+        }
+
+        protected virtual Task<bool> CloseModal(IMvxViewModel viewModel, MvxBasePresentationAttribute attribute)
+        {
+            _windowsContentDialog?.Hide();
```

Comment:
Is Hide correct, or do we really need to remove it as well?

Reply:
I think it's the right one. PrimaryButtonCommand can be invoked, but it's rare.